src/components/Channel/Channel.tsx

```diff
@@ -14,29 +14,208 @@ import DefaultEmojiIndex from 'emoji-mart/dist-modern/utils/emoji-index/nimble-e
+   *
+   * Defaults to null which skips rendering the Channel
+   *
+   * */
+  EmptyPlaceholder?: React.ReactElement;
+  /**
+   * Error indicator UI component. This will be shown on the screen if channel query fails.
+   *
+   * Defaults to and accepts same props as: [LoadingErrorIndicator](https://getstream.github.io/stream-chat-react/#loadingerrorindicator)
+   *
+   * */
+  LoadingErrorIndicator?: React.ComponentType<unknown>; // TODO - add generic when ChannelList pr merged
+  /**
+   * Loading indicator UI component. This will be shown on the screen until the messages are
+   * being queried from channel. Once the messages are loaded, loading indicator is removed from the screen
+   * and replaced with children of the Channel component.
+   *
+   * Defaults to and accepts same props as: [LoadingIndicator](https://github.com/GetStream/stream-chat-react/blob/master/src/components/LoadingIndicator.js)
+   */
+  LoadingIndicator?: React.ComponentType<unknown>; // TODO - add generic when ChannelList pr merged
+  /** Maximum number of attachments allowed per message */
+  maxNumberOfFiles?: number;
+  /**
+   * Message UI component to display a message in message list.
+   *
+   * Available built-in components (also accepts the same props as):
+   *
+   * 1. [MessageSimple](https://github.com/GetStream/stream-chat-react/blob/master/src/components/MessageSimple.js) (default)
+   * 2. [MessageTeam](https://github.com/GetStream/stream-chat-react/blob/master/src/components/MessageTeam.js)
+   * 3. [MessageLivestream](https://github.com/GetStream/stream-chat-react/blob/master/src/components/MessageLivestream.js)
+   * 3. [MessageCommerce](https://github.com/GetStream/stream-chat-react/blob/master/src/components/MessageCommerce.js)
+   *
+   * */
+  Message?: React.ComponentType<unknown>; // TODO - add generic when Message pr merged
+  /** Whether to allow multiple attachment uploads */
+  multipleUploads?: boolean;
+  /**
+   * Handle for click on @mention in message
+   *
+   * @param {Event} event DOM Click event
+   * @param {User} user   Target [user object](https://getstream.io/chat/docs/#chat-doc-set-user) which is clicked
+   */
+  onMentionsClick?: OnMentionAction<Us>;
+  /**
+   * Handle for hover on @mention in message
+   *
+   * @param {Event} event DOM hover event
+   * @param {User} user   Target [user object](https://getstream.io/chat/docs/#chat-doc-set-user) which is hovered
+   */
+  onMentionsHover?: OnMentionAction<Us>;
+};
+
+const UnMemoizedChannel = <
+  At extends UnknownType = DefaultAttachmentType,
+  Ch extends UnknownType = DefaultChannelType,
+  Co extends string = DefaultCommandType,
+  Ev extends UnknownType = DefaultEventType,
+  Me extends UnknownType = DefaultMessageType,
+  Re extends UnknownType = DefaultReactionType,
+  Us extends UnknownType = DefaultUserType
+>(
+  props: PropsWithChildren<ChannelProps<At, Ch, Co, Ev, Me, Re, Us>>,
+) => {
+  const { EmptyPlaceholder = null } = props;
```

Comment:
I see us using `props.channel` or other stuff later referenced off of `props`. Let's just dereference props up here for the values that are used within this file so we can save some on the lookup cost during runtime from doing `props.someVariable`